pyomo/contrib/appsi/base.py

```diff
@@ -1406,17 +1406,17 @@ def update(self, timer: HierarchicalTimer = None):
             vars_to_update = list()
             for v in vars_to_check:
                 _v, lb, ub, fixed, domain_interval, value = self._vars[id(v)]
-                if lb is not v._lb:
-                    vars_to_update.append(v)
-                elif ub is not v._ub:
-                    vars_to_update.append(v)
-                elif (fixed is not v.fixed) or (fixed and (value != v.value)):
+                if (fixed is not v.fixed) or (fixed and (value != v.value)):
```

Comment:
Linters would probably prefer this to be `!=` and not `is not` (because you are comparing `bool`).  The issue is that it is possible to get POD values that compare equal, but are not the same object (so `is not` returns True).  In this case, a false positive it fine (it just means APPSI does a little extra unnecessary work), but it might be better to make a distinction for when we care if an object changes or when the value changes.

Reply:
great point. I'll update this.